share/functions/__fish_cached.fish

```diff
@@ -0,0 +1,35 @@
+function __fish_cached --description "Cache the command output for a given amount of time"
+
+    argparse --min-args 1 --stop-nonopt 't/max-age=!_validate_int --min 0' 'k/cache-key=' -- $argv
+    or return
+
+    if set -q _flag_cache_key
+        set -f cache_key $_flag_cache_key
+    else
+        set -f cache_key (string split ' ' "$argv" | head -1)
+    end
+    if not string match -q --regex '^[\w+-]+$' "$cache_key"
+        return 1
+    end
+
+    if set -q _flag_max_age
+        set -f max_age $_flag_max_age
+    else
+        set -f max_age -1
+    end
+    set -l cache_dir (__fish_make_cache_dir)
```

Comment:
some callers used to have "or return", in case "mkdir" fails; we should probably keep that,
because otherwise we get redundant errors complaining the directory doesn't exist.
It's probably fine to print no completions in that case since it should be a really rare scenario that better be fixed immediately.

Reply:
done